ui/cryptomaterial/dropdown.go

```diff
@@ -228,20 +255,55 @@ func (d *DropDown) Layout(gtx C, dropPos int, reversePos bool) D {
 		iRight = dropPos
 	}
 
+	if d.OpenedLayoutInset == nil {
+		d.OpenedLayoutInset = &layout.Inset{
+			Left:  unit.Dp(float32(iLeft)),
+			Right: unit.Dp(float32(iRight)),
+		}
+	}
+
+	if d.Stack {
+		isOpen := d.isOpen
+		return layout.Stack{Alignment: alig}.Layout(gtx,
+			layout.Expanded(func(gtx C) D {
+				d.isOpen = false // enforce a closed layout display before creating the layout Dimensions and undo later.
+				display := d.closedLayout(gtx, iLeft, iRight)
+				d.isOpen = isOpen
+				return display
+			}),
+			layout.Stacked(func(gtx C) D {
+				maxY := unit.Dp(len(d.items))*values.MarginPadding50 + d.OpenedLayoutInset.Top
+				if d.extraDisplay != nil {
+					maxY += values.MarginPadding50
+				}
+
+				gtx.Constraints.Max.Y = gtx.Dp(maxY)
+				if !d.isOpen {
+					return D{}
+				}
+				return d.openedLayout(gtx)
+			}),
+		)
+	}
+
 	if d.Position == DropdownBasePos && d.isOpenDropdownGroup(d.group) {
+		maxY := unit.Dp(len(d.items)) * values.MarginPadding50
```

Comment:
if d.Stack is true, maxY also adds `d.OpenedLayoutInset.Top` but that is not added here. Intentional?

Reply:
> if d.Stack is true, maxY also adds d.OpenedLayoutInset.Top but that is not added here. Intentional?

Yes, intentional. If d.Stack is true, the dropdown's expanded view is moved below the closed layout view depending on the value set in d.OpenedLayoutInset.Top. That extra shift in vertical space is required to ensure the expanded view has enough space for its elements.